Support Expr.toString and add Json serialization for debugging

jeffmay opened the pull request:
- Add DisplayExpr and make it the toString of Expr
- Add InterpretExprAsJson and InterpretExprResultAsJson for debugging

Depends on:
- [x] https://github.com/jeffmay/vapors/pull/25

kionka-rally commented:
Honestly, this is pretty hard to review with all the codecov warnings. It would be nice if there were an easy to disable them. That said, it all seems reasonable to me.

kionka-rally commented on core/src/test/scala/com/rallyhealth/vapors/factfilter/evaluator/DisplayExprSpec.scala:
```diff
@@ -0,0 +1,49 @@
+        val q = withFactsOfType(FactTypes.Age).where { facts =>
+          facts.toList.map(_.get(_.select(_.value)))
+        }
+        pendingUntilFixed {
```
I'm guessing this PR is a work in progress?

kionka-rally replied:
I'm guessing this PR is a work in progress?

jeffmay committed:
Add class information to DataPath when converting collection types

- Add ClassTag to lens.to() method
- Add ConvertTo DataPath node
- Update ExprBuilder to pass required class tags

jeffmay committed:
Convert Expr.When to use Vector instead of List

- Referencing the branch of a Expr.When condition by index is better supported by a Vector

jeffmay committed:
Add circe serialization of Expr and ExprResult

- Added circe dependencies to the build
- Added InterpretExprAsJson for Expr serialization
- Added InterpretExprResultAsJson for ExprResult serialization
- Created a circe package with standard encoders for facts, fact types and evidence

jeffmay committed:
WIP: Add toString serialization of Expr

- Added DisplayExpr interpreter

jeffmay commented:
Closing this in favor of the new algebra which will replace the code this is based on. The PRs start with https://github.com/jeffmay/vapors/pull/69
